### ISSUE TITLE
[LIB REQUEST]: Add Beman.Optional26 library into Compiler Explorer 

### ISSUE THREAD

**Issue by neatudarius** [compiler-explorer/compiler-explorer#6651]
### Library name

`Beman.Optional26` (`beman_optional26` for static library, if we'll evert deliver a binary; currently, header-only).

### Library description

This request is specifically to integrate a single library from Beman - named `Beman.Optional26` into Compiler Explorer.

### Library repository

https://github.com/beman-project/Optional26

### Motivation

* `Beman.Optional26` is a library within the [Beman Project](https://discourse.boost.org/), used by WG21 to get implementation experience for proposed library features for C++ latest versions.
* The library can be found at [beman-project/Optional26](https://github.com/beman-project/Optional26). Discussion on [discourse](https://discourse.boost.org/).
* This is a header-only library which provides C++26 extensions for std::optional.

* Infra linked issue: [#1348]()https://github.com/compiler-explorer/infra/issues/1348) - PR [#1347](https://github.com/compiler-explorer/infra/pull/1347).
* Beman.Optional26 issue: [#21](https://github.com/beman-project/Optional26/issues/21)

Notes:
* Tested these changes on my local VM, Ubuntu 24.04, arm64 - check video with [local deployment of Beman.Optional26](https://github.com/compiler-explorer/infra/assets/8947836/075c29cf-1417-4110-beb3-caa46c21054d) inside Compiler Explorer:
```shell
compiler-explorer [beman_optional26-trunk] $ diff  etc/config/c++.local.properties etc/config/c++.defaults.properties
156,161c156
< libs=beman_optional26
< libs.beman_optional26.name=Beman.Optional26
< libs.beman_optional26.versions=trunk
< libs.beman_optional26.url=https://github.com/beman-project/Optional26
< libs.beman_optional26.versions.trunk.version=trunk
< libs.beman_optional26.versions.trunk.path=/opt/compiler-explorer/libs/beman_optional26/main/include
---
> libs=
```

**Comment by neatudarius** [comment]
@partouf , when should we expect the deploy after these 2 PRs what were merged? Where can we check? (for future times)

Also, if the deploy pipeline already pulled `beman_optional26` and it's older than commit https://github.com/beman-project/Optional26/commit/da16ab7049fe50696fdd22ce14e0ee8163922743 from `main`, when it will pick up this commit? Note: I've just merged everything we need for tomorrow's WG21 demo with `beman_optional26`.

**Comment by partouf** [comment]
https://godbolt.org/z/9jdGjWsTn

The examples in the readme.md do not work

tag me when you need an update, otherwise somewhere around 3am utc there will be an update

**Comment by neatudarius** [comment]
> https://godbolt.org/z/9jdGjWsTn
> 
> The examples in the readme.md do not work
> 
> tag me when you need an update, otherwise somewhere around 3am utc there will be an update

Now the example that you have attached is working! I suppose you picked an updated example after I pushed a namespace renaming (and godbolt has an old beman_opitonal26).

TLDR: Optional.Beman26 on Compiler Explorer is up and running! Pythagorean Triples: [gcc trunk](https://godbolt.org/z/njhKEqhab) + [clang trunk](https://godbolt.org/z/he1vddGGP)


==========================================================================================

In https://godbolt.org/z/7e8r65WvE I was able to successfully build and run all 5 examples from repo (a.k.a. `examples/*.cpp`) with `-std=C++23` and GCC/Clang trunk:
<img width="2000" alt="godbolt-deploy-test-25 06 2024-beman_optional26-trunk" src="https://github.com/compiler-explorer/compiler-explorer/assets/8947836/6ea47c40-9e42-4993-afc8-458e59c92ab5">

==========================================================================================

This integration is good enough for this WG21 meeting! We'll have few review rounds and be back with comments for versioning bits and check support for multiple copilers / standards.

Thanks @partouf and @mattgodbolt for you help!
